Assets/Mirror/Tests/Editor/WeaverAssembler.cs

```diff
@@ -1,14 +1,38 @@
+using System.Collections.Generic;
 using System.IO;
 using System.Linq;
-using System.Collections.Generic;
+using UnityEditor;
 using UnityEditor.Compilation;
 using UnityEngine;
 
 namespace Mirror.Tests
 {
     public class WeaverAssembler : MonoBehaviour
     {
-        public const string OutputDirectory = "Assets/Mirror/Tests/Editor/WeaverTests~/";
+        static string _outputDirectory;
+        public static string OutputDirectory
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_outputDirectory))
+                {
+                    string[] guidsFound = AssetDatabase.FindAssets($"t:Script WeaverAssembler");
+                    if (guidsFound.Length == 1 && !string.IsNullOrEmpty(guidsFound[0]))
+                    {
+                        string path = AssetDatabase.GUIDToAssetPath(guidsFound[0]);
+                        string dir = Path.GetDirectoryName(path);
+                        // Path in mirror project
+                        //"Assets/Mirror/Tests/Editor/WeaverTests~/";
+                        _outputDirectory = Path.Combine(dir, "WeaverTests~/");
```

Comment:
so it still needs to be named "WeaverTests~"?

Reply:
the ~ excludes it from unity, so it is still needed.

Reply:
@James-Frowen what I meant was: do we still need to have the folder name hardcoded, we can't auto detect it somehow?

Reply:
I think we still need it